### PR TITLE
Add comparison matrix to Velocity comparison page

### PR DESCRIPTION
**Pull request by Malfrador**
A question that gets asked often is "Should I use Velocity, Waterfall or BungeeCord". While the existing page already answer that, its a lot to read and contains plenty of details that most server owners just looking for a simple answer aren't interested in.

This PR adds a small markdown table with a comparison matrix to the top of that doc page. It makes it easy to see at a glance what Velocity does, and what it doesn't do - mainly, supporting plugins made for BungeeCord, another question that gets asked very often. 

chrome_PdzuRM5Us1

Goal of this PR is to have an easy to understand overview for someone just looking for a simple answer to the aforementioned question. Those interested in more in-depth information can just continue reading the docs page. 
With the table added, the docs page can simply be linked everytime someone asks that question, and contains a quick answer for them.

### CODE REVIEW

File: docs/velocity/admin/reference/comparison.md [hunk 1]
```diff
@@ -8,6 +8,16 @@ Presumably, you have learned about what Velocity can do for you. But how does it
 other solutions out there? We're trying to convince you to use Velocity, after all, so this document
 may be somewhat biased in our favor.
 
+## Overview
+| Feature                           	   | BungeeCord | Waterfall | Velocity |
+|------------------------------------------|------------|-----------|----------|
+| BungeeCord plugins                       | ✅          | ✅         | ❌        |
+| Velocity plugins                         | ❌          | ❌         | ✅        |
+| Secure player information forwarding     | ❌          | ❌         | ✅        |
+| API supporting modern Minecraft features | ❌          | ❌         | ✅        |
```
Comment (Nacioszeczek) [inline docs/velocity/admin/reference/comparison.md:17]: Could list some of the stuff, e.g. Adventure support

File: docs/velocity/admin/reference/comparison.md [hunk 2]
```diff
@@ -8,6 +8,16 @@ Presumably, you have learned about what Velocity can do for you. But how does it
...
+| Actively developed                       | ❓          | ❌         | ✅        |
+| Improved mod support                     | ❌          | ❌         | ✅        |
```
Comment (Nacioszeczek) [inline docs/velocity/admin/reference/comparison.md:19]: Should probably mention modern Forge being problematic, on all proxies though

File: docs/velocity/admin/reference/comparison.md [hunk 3]
```diff
@@ -8,6 +8,16 @@ Presumably, you have learned about what Velocity can do for you. But how does it
 other solutions out there? We're trying to convince you to use Velocity, after all, so this document
 may be somewhat biased in our favor.
 
+## Overview
+| Feature                           	   | BungeeCord | Waterfall | Velocity |
+|------------------------------------------|------------|-----------|----------|
+| BungeeCord plugins                       | ✅          | ✅         | ❌        |
```
Comment (Nacioszeczek) [inline docs/velocity/admin/reference/comparison.md:14]: It is somewhat possible to run Bungee plugins on Velocity using a plugin, not sure if that's reliable enough to mention that on the docs though

**Review by Nacioszeczek — CHANGES_REQUESTED** [review]

**Comment by olijeffers0n** [comment]
Two rows for velocity and bungee plugins seem redundant but i can’t think of a better way to do this. Also, surely there must be another pro to Bungee over velocity? Seems rather one sided that’s all 😄

**Comment by MiniDigger** [comment]
that table should somehow also show a difference between bungee and waterfall (not that I would know something to add there, lol)

**Comment by A248** [comment]
If you haven't any differences between BungeeCord and Waterfall, or you aren't willing to document them, you may want to remove one of the columns entirely.

There are differences. Waterfall guarantees Slf4j and exposes it in the API. On BungeeCord this library happens to be provided, but only due to the recent circumstance of library loading. Moreover, Waterfall has miscellaneous performance improvements to the implementation. I believe there might be a few new events. If we can get https://github.com/PaperMC/Waterfall/pull/812 merged, then we can also say BungeeCord doesn't address some thread safety bugs -- which would be true.

By the way, Velocity can run some BungeeCord plugins via [Snap](https://github.com/Phoenix616/Snap). You may want to consider, thus, a mixed marker between a checkmark and an X.

Finally, in Velocity's favor, I strongly suggest adding a "Recommended by A248" row to your comparison matrix.

File: docs/velocity/admin/reference/comparison.md [hunk 4]
```diff
@@ -8,6 +8,23 @@ Presumably, you have learned about what Velocity can do for you. But how does it
 other solutions out there? We're trying to convince you to use Velocity, after all, so this document
 may be somewhat biased in our favor.
 
+## Overview
+This is a quick overview of the differences between Velocity and other popular proxy software. Read below for more details.
+| Feature                           	   | BungeeCord | Waterfall | Velocity |
+|------------------------------------------|------------|-----------|----------|
+| Resource efficient					   | ❌         | ✅        | ✅✅     |
```
Comment (olijeffers0n) [inline docs/velocity/admin/reference/comparison.md:15]: Tabs not spaces

File: docs/velocity/admin/reference/comparison.md [hunk 5]
```diff
@@ -8,6 +8,23 @@ Presumably, you have learned about what Velocity can do for you. But how does it
...
+| Velocity plugins                         | ❌         | ❌        | ✅       |
+| BungeeCord plugins                       | ✅         | ✅        | 🟨*      |
+| Secure player information forwarding     | ❌         | ❌        | ✅       |
+| API supporting modern Minecraft features | ❌         | ❌        | ✅       |
+| Actively developed                       | ❓         | ❌        | ✅       |
+| Improved mod support                     | ❌         | ❌        | ✅*\*    |
+
+\* _The Velocity API does not support plugins made for BungeeCord/Waterfall, but [Snap](https://hangar.papermc.io/Phoenix616/Snap) can be installed for experimental support.
+ Snap is not maintained by the Velocity project._
```
Comment (olijeffers0n) [inline docs/velocity/admin/reference/comparison.md:24]:
```suggestion
 Snap is not maintained by, or affiliated with, the Velocity project._
```

**Review by olijeffers0n — CHANGES_REQUESTED** [review]

**Review by olijeffers0n — APPROVED** [review]

**Review by Nacioszeczek — APPROVED** [review]

**Review by Warriorrrr — APPROVED** [review]

**Pull request merged**